### PR TITLE
docs: add mention of bors to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by couchand**
As of #24100 we can use Bors to merge PRs to master, and once
#22499 is complete that will be the standard practice.  This
adds documentation to make this clear.

Contributes to #22499
Release note: None

### CODE REVIEW

**Review by benesch — APPROVED** [review]
LGTM, though I worry that the change describes how things will be in two weeks rather than how they are now.

**Comment by couchand** [comment]
> LGTM, though I worry that the change describes how things will be in two weeks rather than how they are now.

I'm fine waiting on merging this until shortly before the kickoff/demo/whatever.

> Actually, if you have a free second, this PR should update first-pr.md too.

Oh yeah, I started to do that, but got distracted with other stuff.

**Review by benesch — APPROVED** [review]

**Comment by benesch** [comment]
Thanks!!

File: docs/first-pr.md [hunk 1]
```diff
@@ -573,21 +573,17 @@ in doubt, ask!
...
-that "squashes" together the nicely-curated logical changes and messages from
-the PR's constituent commits. If your PR has fixup commits to address review
-feedback, or generally sloppy commits, you should rebase those into a more
-logical commit series locally and force-push.
+revert at least one PR that you land in your first three months.
+
+You shouldn't merge yourself, instead, request our build bot Craig to do it for
+you.  Craig is a [Bors merge bot], whose only job is to be gatekeeper to the
+repository.  When you're ready to merge, add a comment on the PR of the form
+`bors r=<reviewer>`, replacing `<reviewer>` with the person who gave you the
```
Comment (jordanlewis) [inline docs/first-pr.md]: s/person/GitHub username of the person/

File: docs/first-pr.md [hunk 2]
```diff
@@ -573,21 +573,17 @@ in doubt, ask!
...
-
-Make sure to use the default "Create a merge commit" mode when you hit the merge
-button. If you instead use "Squash and merge," you'll end up with one commit
-that "squashes" together the nicely-curated logical changes and messages from
-the PR's constituent commits. If your PR has fixup commits to address review
-feedback, or generally sloppy commits, you should rebase those into a more
-logical commit series locally and force-push.
+revert at least one PR that you land in your first three months.
+
+You shouldn't merge yourself, instead, request our build bot Craig to do it for
```
Comment (jordanlewis) [inline docs/first-pr.md]: suggest different wording: "Finally, to merge your PR, request a merge from our built bot Craig. Craig is ..."

File: docs/first-pr.md [hunk 3]
```diff
@@ -573,21 +573,17 @@ in doubt, ask!
...
+you.  Craig is a [Bors merge bot], whose only job is to be gatekeeper to the
+repository.  When you're ready to merge, add a comment on the PR of the form
```
Comment (jordanlewis) [inline docs/first-pr.md]: Single spaces between sentences to match this doc and the rest of our comment and prose style plz

File: docs/first-pr.md (hunk 2 above)
Comment (jordanlewis) [reply docs/first-pr.md]: er, s/built/build/

File: docs/first-pr.md (hunk 1 above)
Comment (couchand) [reply docs/first-pr.md]: done.

File: docs/first-pr.md (hunk 2 above)
Comment (couchand) [reply docs/first-pr.md]: made a few edits based on your suggestion, trying to work it better into the surrounding context, PTAL

File: docs/first-pr.md (hunk 3 above)
Comment (couchand) [reply docs/first-pr.md]: 😿

**Comment by jordanlewis** [comment]
LGTM!

**Comment by knz** [comment]
plz merge this

**Comment by couchand** [comment]
bors r=benesch,jordanlewis

**Comment by knz** [comment]
bors r=couchand

**Comment by knz** [comment]
I rebased to pick up this morning's URL change

**Comment by knz** [comment]
bors r=couchand

**Comment by knz** [comment]
I'm going to merge this manually until we figure out what's keeping bors awake at night.

**Pull request merged**